python/desc/imsim/ImageSimulator.py

```diff
@@ -425,27 +442,44 @@ def update_checkpoint_summary(self, gs_interpreter, num_objects):
             self.sender.send((nobjs, self.sensor_name, num_objects,
                               gs_interpreter.nobj_checkpoint))
 
-
-    def write_raw_files(self, detector_images):
+    def write_raw_files(self, gs_interpreter):
         """
         Write the raw files directly from galsim images.
 
         Parameters
         ----------
-        detector_images: dict
-            The dictionary attribute of the GalSimInterpreter object
-            used to fill the galsim images with eimage data.
+        gs_interpreter: GalSimInterpreter object
         """
         persist = IMAGE_SIMULATOR.config['persistence']
-        prefix = persist['raw_file_prefix']
+        band = IMAGE_SIMULATOR.obs_md.bandpass
+        for detector in gs_interpreter.detectors:
+            filename = gs_interpreter._getFileName(detector, band)
+            try:
+                gs_image = gs_interpreter.detectorImages[filename]
+            except KeyError:
+                continue
+            else:
+                raw = ImageSource.create_from_galsim_image(gs_image)
+                outfile = IMAGE_SIMULATOR.output_file(detector.name, raw=True)
+                raw.write_fits_file(outfile,
+                                    compress=persist['raw_file_compress'])
+
+    def write_eimage_files(self, gs_interpreter):
+        """
+        Write the eimage files.
+
+        Parameters
+        ----------
+        gs_interpreter: GalSimInterpreter object
+        """
+        prefix = IMAGE_SIMULATOR.config['persistence']['eimage_prefix']
         obsHistID = str(IMAGE_SIMULATOR.obs_md.OpsimMetaData['obshistID'])
         nameRoot = os.path.join(IMAGE_SIMULATOR.outdir, prefix) + obsHistID
```

Comment:
This is OK but I see in the write_raw_file function above you are leveraging the IMAGE_SIMULATOR.output_file function to build this path.  Is there are reason it has to be done manually here?

Reply:
Yes.  The sims_GalSimInterface code composes the output filenames given nameRoot not the actual filename, so output_file can't be used here.